modules/highgui/src/window_cocoa.mm

```diff
@@ -161,162 +146,145 @@ CV_IMPL int cvInitSystem( int , char** )
     if( floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_5 )
         [application setActivationPolicy:NSApplicationActivationPolicyRegular];
 #endif
-    //[application finishLaunching];
-    //atexit(icvCocoaCleanup);
-
     setlocale(LC_NUMERIC,"C");
 
     return 0;
 }
 
 static CVWindow *cvGetWindow(const char *name) {
-    //cout << "cvGetWindow" << endl;
-    NSAutoreleasePool* localpool = [[NSAutoreleasePool alloc] init];
-    NSString *cvname = [NSString stringWithFormat:@"%s", name];
-    CVWindow* retval = (CVWindow*) [windows valueForKey:cvname] ;
-    //cout << "retain count: " << [retval retainCount] << endl;
-    //retval = [retval retain];
-    //cout << "retain count: " << [retval retainCount] << endl;
-    [localpool drain];
-    //cout << "retain count: " << [retval retainCount] << endl;
-    return retval;
+    @autoreleasepool{
```

Comment:
Please refer to the following guide on how to return object managed by autorelease pools: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmAutoreleasePools.html
In particular, the example with `findMatchingObject` which is close to the use case